staging/src/k8s.io/client-go/rest/request.go

```diff
@@ -152,6 +154,9 @@ func NewRequest(c *RESTClient) *Request {
 	case len(c.content.ContentType) > 0:
 		r.SetHeader("Accept", c.content.ContentType+", */*")
 	}
+
+	// generate an audit ID
+	r.SetHeader("Audit-ID", uuid.New().String())
```

Comment:
I feel like this whole cross-component topic makes the change much bigger, i.e. crossing the line to need an enhancement.

Reply:
yes, I removed the client-go changes. I will create a new KEP with the proposed client-go changes.